ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -193,14 +195,18 @@ PrjFS_Result PrjFS_StartVirtualizationInstance(
         << callbacks.GetFileStream << ", "
         << callbacks.NotifyOperation << ", "
         << callbacks.LogError << ","
+        << callbacks.LogWarning << ","
+        << callbacks.LogInfo << ","
         << poolThreadCount << ")" << endl;
 #endif
     
     if (nullptr == virtualizationRootFullPath ||
         nullptr == callbacks.EnumerateDirectory ||
         nullptr == callbacks.GetFileStream ||
         nullptr == callbacks.NotifyOperation ||
-        nullptr == callbacks.LogError)
+        nullptr == callbacks.LogError ||
+        nullptr == callbacks.LogWarning ||
+        nullptr == callbacks.LogInfo)
     {
         return PrjFS_Result_EInvalidArgs;
     }
```

Comment:
There are several error conditions in `PrjFS_StartVirtualizationInstance` with no logging where it would be helpful.

Here are two examples, could you add logging here and anywhere else that errors would cause `PrjFS_StartVirtualizationInstance` to fail?

```
    DataQueueResources dataQueue;
    s_messageQueueDispatchQueue = dispatch_queue_create("PrjFS Kernel Message Handling", DISPATCH_QUEUE_SERIAL);
    if (!PrjFSService_DataQueueInit(&dataQueue, s_kernelServiceConnection, ProviderPortType_MessageQueue, ProviderMemoryType_MessageQueue, s_messageQueueDispatchQueue))
    {
        cerr << "Failed to set up shared data queue.\n";
        return PrjFS_Result_EInvalidOperation;
    }
    
    s_virtualizationRootFullPath = virtualizationRootFullPath;
    s_callbacks = callbacks;
    
    errno_t error = RegisterVirtualizationRootPath(virtualizationRootFullPath);
    if (error != 0)
    {
        cerr << "Registering virtualization root failed: " << error << ", " << strerror(error) << endl;
        return PrjFS_Result_EInvalidOperation;
    }
```